src/backend/distributed/master/master_metadata_utility.c

```diff
@@ -62,6 +64,246 @@ static ShardPlacement * TupleToShardPlacement(TupleDesc tupleDesc,
+				 workerNodePort) ||
+				distributedTableCacheEntry->partitionMethod == DISTRIBUTE_BY_NONE)
+			{
+				uint64 shardId = placement->shardId;
+				char *shardName = get_rel_name(distributedRelationId);
+				char *shardQualifiedName = NULL;
+				char *quotedShardName = NULL;
+				AppendShardIdToName(&shardName, shardId);
+
+				shardQualifiedName = quote_qualified_identifier(schemaName, shardName);
+				quotedShardName = quote_literal_cstr(shardQualifiedName);
+
+				if (!shardSeenOnNode)
+				{
+					shardSeenOnNode = !shardSeenOnNode;
+				}
+				else
+				{
+					appendStringInfo(tableSizeQuery, " + ");
+				}
+
+				if (sizeFunctionType == CITUS_TOTAL_RELATION_SIZE)
+				{
+					appendStringInfo(tableSizeQuery, "pg_total_relation_size(%s)",
+									 quotedShardName);
+				}
+				else if (sizeFunctionType == CITUS_TABLE_SIZE)
+				{
+					appendStringInfo(tableSizeQuery, "pg_table_size(%s)",
+									 quotedShardName);
+				}
+				else if (sizeFunctionType == CITUS_RELATION_SIZE)
+				{
+					appendStringInfo(tableSizeQuery, "pg_relation_size(%s)",
+									 quotedShardName);
+				}
+
+				/*For reference table, there is only one shard placement for each node */
+				if (distributedTableCacheEntry->partitionMethod == DISTRIBUTE_BY_NONE)
+				{
+					appendStringInfo(tableSizeQuery, ";");
+					break;
+				}
+			}
+		}
+
+		/* There is no shard placement for this node */
+		if (!shardSeenOnNode)
+		{
+			continue;
+		}
+
+		appendStringInfo(tableSizeQuery, ";");
+
+
+		connection = GetNodeConnection(0, workerNodeName, workerNodePort); /* TODO : I'm not sure about the flag ! */
+
+		executeCommand = ExecuteOptionalRemoteCommand(connection,
+													  tableSizeQuery->data,
+													  &queryResult);
+
+		/* If there is a connection error and the table is not reference table,
+		 * close the connection and quit. If it is a reference table continue
+		 * to get the size from another node. */
+		if (executeCommand != 0)
+		{
+			PQclear(queryResult);
+
+			ForgetResults(connection);
+
+			if (distributedTableCacheEntry->partitionMethod == DISTRIBUTE_BY_NONE)
+			{
+				continue;
+			}
+
+			ereport(ERROR, (errcode(ERRCODE_CONNECTION_FAILURE),
+							errmsg("cannot get the size")));
+		}
+
+		tableSizeString = PQgetvalue(queryResult, 0, 0);
+		totalRelationSize += atoi(tableSizeString);
+
+		PQclear(queryResult);
+
+		ForgetResults(connection);
+
+		/*Size of single shard is enough to find the size of reference table*/
+		if (distributedTableCacheEntry->partitionMethod == DISTRIBUTE_BY_NONE)
+		{
+			return totalRelationSize * list_length(workerNodeList);
+		}
+	}
+
+	/*Size must be returned inside the loop for reference tables*/
+	if (distributedTableCacheEntry->partitionMethod == DISTRIBUTE_BY_NONE)
+	{
+		ereport(ERROR, (errcode(ERRCODE_CONNECTION_FAILURE),
+						errmsg("cannot get the size")));
+	}
+
+	return totalRelationSize;
+}
+
+
+/*
+ * citus_total_relation_size accepts a table name and
```

Comment:
You can continue to write function comment up to 90 characters before continuing from the next line.